guides/2022-12-15-stream-inspector.mdx

```diff
@@ -0,0 +1,54 @@
+---
+title: 'Stream Inspector'
+slug: stream-inspector
+tag: ['Stream Inspector']
+description: Learn how to use the stream inspector
+keywords: ['stream', 'inspector', 'conduit']
+---
+
+In this guide, we are going to learn how use Conduit's stream inspector. Stream inspection is available via the Conduit UI and the API.
+
+## UI
+
+To access the stream inspector through the UI, first navigate to the pipeline which you'd like to inspect. Then, click on the connector in which you're interested. You'll see something similar to this:
+
+![stream inspector pipeline view](/images/stream-inspector-pipeline.png)
+
+Click the "Inspect Stream" button to start inspecting the connector. A new pop-up window will show the records:
+
+![stream inspector show stream](/images/stream-inspector-show-stream.png)
+
+On the "Stream" tab you'll see the latest 10 records. If you switch to the "Single record" view, only the last record will be shown. You can use the "Pause" button to pause the inspector and stop seeing the latest record(s). The ones that are already shown will be kept so you can inspect them more thoroughly.
+
+## API
+
+To access the stream inspector through the API, you'll need a WebSocket client. [wscat](https://github.com/websockets/wscat) is one of them. The URL on which the inspector is available comes in the following format: `ws://host:port/v1/connectors/<connector ID>/inspect`. For example, if you have Conduit installed locally and with the default settings, you can inspector a connector by running the following command:
```

Comment:
We could also note that the fields `key`, `payload.after`, `payload.before` and `position` can contain raw binary data that's base64 encoded, which can be decoded using `jq` (that's pretty useful IMO).